utils.py

```diff
@@ -742,18 +818,19 @@ def require_valid_page_title_fragment_for_web(page_title_fragment_for_web):
         ValidationError. Page title fragment is too lengthy.
     """
     max_chars_in_page_title_frag_for_web = (
-        constants.MAX_CHARS_IN_PAGE_TITLE_FRAGMENT_FOR_WEB)
+        constants.MAX_CHARS_IN_PAGE_TITLE_FRAGMENT_FOR_WEB) # type: ignore[attr-defined]
     if not isinstance(page_title_fragment_for_web, python_utils.BASESTRING):
         raise ValidationError(
             'Expected page title fragment to be a string, received %s'
             % page_title_fragment_for_web)
     if len(page_title_fragment_for_web) > max_chars_in_page_title_frag_for_web:
         raise ValidationError(
             'Page title fragment should not be longer than %s characters.'
-            % constants.MAX_CHARS_IN_PAGE_TITLE_FRAGMENT_FOR_WEB)
+            % constants.MAX_CHARS_IN_PAGE_TITLE_FRAGMENT_FOR_WEB) # type: ignore[attr-defined]
 
 
 def capitalize_string(input_string):
+    # type: (Optional[Text]) -> Optional[Text]
```

Comment:
empty string is not optional, if `None` was allowed then it would be optional
```suggestion
    # type: (Text) -> Text
```

Reply:
In our test `test_capitalize_string()`, we also pass None to it. So I think Optional would be a better choice here as it must have been designed to accept None. What do you think?

Reply:
If `None` is passed just in tests then remove that test. This is a rule to follow in general, do not conform the types to tests but tests to types. A slight exception might be data coming from 'unknown' sources like HTTP requests.

Reply:
For now this is used only at one place to capitalize a string whose default is `''`. So I think Removing optional will work.

Reply:
Done